File: 202_Happy_Number/solution.c
int sum_of_squares(int n) {
    int sum = 0, digit = 0;
    while (n > 0) {
        digit = n % 10;
        sum += (digit * digit);
        n /= 10;
    }
    return sum;
}
bool isHappy(int n) {
    int slow = n, fast = n;
    do {
        slow = sum_of_squares(slow);
        fast = sum_of_squares(sum_of_squares(fast));
    } while (slow != fast);
    return slow == 1;
}

File: 202_Happy_Number/solution.py
class Solution:
    def sum_of_digit(self, num):
        total = 0
        while num > 0:
            digit = num % 10
            total += digit**2
            num //= 10
        return total

    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1 and n not in seen:
            seen.add(n)
            n = self.sum_of_digit(n)
        return n == 1
